server/rpc.go

```diff
@@ -67,6 +67,59 @@ func genTranscoderReq(b Broadcaster, jid int64) (*net.TranscoderRequest, error)
 	return &net.TranscoderRequest{JobId: jid, Sig: sig}, nil
 }
 
+func checkTranscoderAvailability(orch Orchestrator) {
+	time.Sleep(1)
+	ts := time.Now()
+	ts_signature, err = orch.Sign([]byte(fmt.Sprintf("%v", ts)))
+	if err != nil {
+		return false, err
+	}
+
+	ping, err := crypto.Keccak256(ts_signature)
+	if err != nil {
+		return false, err
+	}
+
+	orch_client, err = startOrchestratorClient(orch.addr)
+	if err != nil {
+		return false, err
+	}
+
+	pong, err = orch_client.Ping(&PingPong{value: ping})
+	if err != nil {
+		return false, err
+	}
+
+	// is a signature the same as an address? is this the verify function we need here?
+	return eth.VerifySig(pong, ping, orch.addr), nil
+}
+
+func startOrchestratorClient(orchestratorServer string) error {
```

Comment:
Excellent -- do you think we can also take the opportunity to refactor StartBroadcaster to share this function?

Reply:
Absolutely!